src/Provider/ImageProvider.php

```diff
@@ -207,6 +208,49 @@ public function generatePrivateUrl(MediaInterface $media, $format)
         return $this->thumbnail->generatePrivateUrl($this, $media, $format);
     }
 
+    public function validate(ErrorElement $errorElement, MediaInterface $media)
+    {
+        parent::validate($errorElement, $media);
+
+        if (null === $media->getBinaryContent()) {
+            return;
+        }
+
+        $minWidth = 0;
+        $minHeight = 0;
+
+        foreach ($this->getFormatsForContext($media->getContext()) as $format) {
+            if (!$format['constraint']) {
+                continue;
+            }
+
+            $minWidth = max($minWidth, $format['width'] ?? 0);
+            $minHeight = max($minHeight, $format['height'] ?? 0);
+        }
+
+        if (0 === $minWidth && 0 === $minHeight) {
+            return;
+        }
+
+        try {
+            $image = $this->imagineAdapter->open($media->getBinaryContent()->getPathname());
+        } catch (\RuntimeException $e) {
+            return;
```

Comment:
Shouldn't we log something here? Or at least add a comment?

Reply:
It's a copy & paste of https://github.com/sonata-project/SonataMediaBundle/blob/8dd4990922efbc4163f528b77a6a677f3027f1bc/src/Provider/ImageProvider.php#L235-L239

If we can an error, we would trigger a validation error in an other component.

Reply:
> If we can an error, we would trigger a validation error in an other component.

Sorry, can you fix that sentence please? I don't understand it.

Reply:
Added a comment in the code